WordPress/Sniffs/PHP/EnqueuedCheckSniff.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\PHP;
+
+use WordPress\AbstractFunctionParameterSniff;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * Based off the StrictInArraySniff.php sniff, this checks the enqueued 4th parameter to make sure a version is available.
+ * The Enqueued functions are:
+ * wp_register_script()
+ * wp_enqueue_script()
+ * wp_register_style()
+ * wp_enqueue_style()
+ * If a source ($src) value is passed, then version ($ver) needs to have non-falsy value.
+ * If a source ($src) value is passed a check for in footer ($in_footer), warn the user if the value is falsy.
+ *
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1146
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   1.0.0
+ */
+class EnqueuedCheckSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 1.0.0
+	 * @var string
+	 */
+	protected $group_name = 'Enqueued';
+
+	/**
+	 * List of enqueued functions that need to be check to make sure
+	 *
+	 * @link https://developer.wordpress.org/reference/functions/wp_register_script/
+	 * @link https://developer.wordpress.org/reference/functions/wp_enqueue_script/
+	 * @link https://developer.wordpress.org/reference/functions/wp_register_style/
+	 * @link https://developer.wordpress.org/reference/functions/wp_enqueue_style/
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array <string function_name> => <bool always needed ?>
+	 */
+	protected $target_functions = array(
+		'wp_register_script' => true,
+		'wp_enqueue_script'  => true,
+		'wp_register_style'  => true,
+		'wp_enqueue_style'   => true,
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		// Check to see if a source ($src) is specified.
+		if ( ! isset( $parameters[2] ) ) {
+			return;
+		}
+
+		/*
+		 * Version Check
+		 *
+		 * Check to make sure the version is set explicitly.
+		 */
+
+		if ( ! isset( $parameters[4] ) || $parameters[4]['raw'] === 'null' ) {
+			$this->phpcsFile->addError(
+				'Version not set for %s.',
```

Comment:
What about something along the lines of: `Script/Style version not set in call to %s(). This means new versions of the script/style will not always be loaded due to browser caching.`